test/jdk/java/awt/event/MouseEvent/AWTPanelSmoothWheel.java

```diff
@@ -25,50 +25,82 @@
+    private static int hiResWheelCount = 0;
+    private static final String INSTRUCTIONS = """
+            <html>
+            <body>
+            This test is relevant on platforms with high-resolution mouse wheel,
+            please press PASS if this is not the case.<br> <br>
 
             Place the mouse cursor above the green panel and rotate the mouse wheel,
-            the test will print all mouse wheel event messages into the logging panel
-            below the instruction window.
-            Please make sure that some of the messages have non-zero 'wheelRotation' value,
-            and also check if the test works OK if the mouse wheel is rotated very slow.
+            the test will print mouse wheel event messages in the format
+            <b> [Event#, WheelRotation, PreciseWheelRotation]</b> into the logging
+            panel below the instruction window.<br> <br>
 
-            If the above is true press PASS, else FAIL.
+            A hi-res mouse is one which produces MouseWheelEvents having
+            <b>preciseWheelRotation &lt 1.</b> <br>
+            When preciseWheelRotation adds up to 1,wheelRotation becomes 1. <br>
+            You should see a few events where preciseWheelRotation < 1 & wheelRotation = 0
+            followed by a event where preciseWheelRotation = 1 & wheelRotation = 1.<br> <br>
+
+            Check if the test works OK when the mouse wheel is rotated very slow.<br> <br>
+            Please press PASS if above is true, else FAIL. <br> <br>
+
+            <hr>
+            PS: If you don't see events with preciseWheelRotation < 1,
+            then the mouse doesn't support high-resolution scrolling.
+            A warning is shown in the logging area if you are not using a hi-res mouse.
+            <br> <br>
+            </body>
+            </html>
             """;
 
-    public static void main (String[] args) throws Exception {
+    public static void main(String[] args) throws Exception {
         PassFailJFrame.builder()
                 .title("Test Instructions")
                 .instructions(INSTRUCTIONS)
-                .rows((int) INSTRUCTIONS.lines().count() + 2)
-                .columns(45)
-                .logArea(8)
+                .rows(20)
+                .columns(50)
+                .testTimeOut(10)
+                .logArea(10)
                 .testUI(AWTPanelSmoothWheel::createUI)
                 .build()
                 .awaitAndCheck();
     }
 
-    private static Frame createUI () {
+    private static Frame createUI() {
         Frame frame = new Frame("Test Wheel Rotation");
         Panel panel = new Panel();
         panel.setBackground(Color.GREEN);
-        panel.addMouseWheelListener(e -> PassFailJFrame.log(e.toString()));
-        frame.setSize (200, 200);
+        panel.addMouseWheelListener(e -> {
+            if (e.getScrollType() == MouseWheelEvent.WHEEL_UNIT_SCROLL) {
+                PassFailJFrame.log("WheelEvent#"+ ++wheelEventCount
```

Comment:
```suggestion
                PassFailJFrame.log("WheelEvent#" + (++wheelEventCount)
```
Use parentheses to avoid confusion.